src/gitlab/services/gitlab-api/pipelines-api.ts

```diff
@@ -58,17 +58,24 @@ export namespace GitlabPipelines {
         return result.promise;
     }
 
-    export function getPipelinesForMR(mergeRequestId: number) {
+    export async function getPipelinesForMR(mergeRequestId: number) {
         const projectId = getProjectId();
-        const result = new Deferred<PipelineBase[]>();
 
-        $.ajax(`/api/v4/projects/${projectId}/merge_requests/${mergeRequestId}/pipelines`)
-            .then((data: PipelineScheduleBase[]) => {
-                result.resolve(data);
-            })
-            .catch(result.reject);
-
-        return result.promise;
+        try {
+            const mrData = await $.ajax(`/api/v4/projects/${projectId}/merge_requests/${mergeRequestId}`);
```

Comment:
Also, I think these files are split according to Gitlab API. I would place it to separate file. According to the [Gitlab API](https://docs.gitlab.com/ee/api/merge_requests.html#list-mr-pipelines), this should belong to Merge Request API section.

But I would leave this up to the repo author, @qmOnArq.

Reply:
Removed the request